src/content/reference/rules/components-and-hooks-must-be-pure.md

```diff
@@ -0,0 +1,282 @@
+---
+title: Components and Hooks must be pure
```

Comment:
It might make this page easier to write and understand if you had a small section at the beginning that desribes what purity is, why its important and kicks people off to the /learn/keeping-components-pure page for a refresher and have the rest of the page focused on more philosophy of why components should be pure. Here is an sample high level outline:

What is pure component?
Idempotency
Side effects
Immutability

Right now this page feels like its entangled with /learn/keeping-components-pure and I'm not sure when I would send someone to this page vs /learn/keeping-components-pure. I think creating a very accessible summary, then listing /learn/keeping-components-pure as a "prerequesite" to the rest of the page will allow this page to have its own purpose and identity distinct from /learn/keeping-components-pure.